### ISSUE TITLE
Provide better support to configure logging backends via application configuration

### ISSUE THREAD

**Issue by tsachev** [spring-projects/spring-boot#4032]
Currently only log levels and log files can be configured via application properties/yaml/json.
It would be nice to be able to specify any properties to the underlying logging system.
This should be doable. I think all supported logging systems but, logback can be configured via properties files (including log4j2 since version 2.4).

Something like this

```
logging.log4j2.log4j2specifproperty=value
logging.log4j.log4jspecifproperty=value
logging.jul.julspecifproperty=value
```

**Comment by wilkinsona** [comment]
I think this duplicates #1788 which was implemented in 1.3.0.M2, for Logback at least. See [the documentation](http://docs.spring.io/spring-boot/docs/current-SNAPSHOT/reference/htmlsingle/#_environment_properties) for more details.

**Comment by snicoll** [comment]
Duplicates #1788
